autorest/validation/validation.go

```diff
@@ -0,0 +1,381 @@
+/*
+Package validation provides methods for validating parameter value using reflection.
+*/
+package validation
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+// Constraint stores constraint name, target field name
+// Rule and chain validations.
+type Constraint struct {
+
+	// Target field name for validation.
+	Target string
+
+	// Constraint name e.g. minLength, maxLength, pattern, etc.
+	Name string
```

Comment:
so since these field is `string`, will the sdk say `Constraint{"fieldName", "Empty", ...}`? Assuming the answer is yes, instead of making SDK say `"Empty"` can't we export the constraint consts below and refer to it as `constraint.Empty` in the SDK code?